### PR TITLE
feat: add react native enhancer

### PR DESCRIPTION
**Pull request by satazor**

### CODE REVIEW

File: lib/enhancers/with-react-native/index.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+module.exports = (config) => {
+    const reactNativePreset = require('react-native/jest-preset');
+
+    config.testEnvironment = reactNativePreset.testEnvironment;
+
+    config.haste = {
+        ...config.haste,
+        ...reactNativePreset.haste,
+    };
```
Comment (acostalima) [inline lib/enhancers/with-react-native/index.js]: Jest has a default configuration for `haste`, but as React Native's preset does make use of it I'm assuming there's a good reason to do so. I don't know what are the implications, but I'd suggest to strictly stick to React Native's config only.

```suggestion
    config.haste = reactNativePreset.haste;
```

FYI, the default config is:

```js
haste: {
   computeSha1: false,
   providesModuleNodeModules: [],
   throwOnModuleCollision: false
}
```

Comment (satazor) [reply lib/enhancers/with-react-native/index.js]: Doing what you are doing or sticking to react native is the same, as Jest will end up mixing their default config with react native. In the end, `haste` will contain all defaults + react-native in both scenarios.

Comment (acostalima) [reply lib/enhancers/with-react-native/index.js]: Is that so? That means a deep merge is performed by Jest then? If that's the case then so be it. 😄

**Comment by satazor** [comment]
Superseded by #16

**Pull request closed without merging**